rules/TypeDeclaration/Rector/FunctionLike/AddParamTypeForFunctionLikeWithinCallLikeArgFromArgDeclarationRector.php

```diff
@@ -0,0 +1,273 @@
+{
+    /**
+     * @var AddParamTypeForFunctionLikeWithinCallLikeArgFromArgDeclaration[]
+     */
+    private array $addParamTypeForFunctionLikeParamDeclarations = [];
+
+    private bool $hasChanged = false;
+
+    public function __construct(
+        private readonly TypeComparator $typeComparator,
+        private readonly PhpVersionProvider $phpVersionProvider,
+        private readonly StaticTypeMapper $staticTypeMapper,
+    ) {
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition('Add param types where needed', [
+            new ConfiguredCodeSample(
+                <<<'CODE_SAMPLE'
+$app->extend(SomeClass::class, function ($parameter) {});
+CODE_SAMPLE
+                ,
+                <<<'CODE_SAMPLE'
+$app->extend(SomeClass::class, function (SomeClass $parameter) {});
+CODE_SAMPLE
+                ,
+                [
+                    new AddParamTypeForFunctionLikeWithinCallLikeArgFromArgDeclaration(
+                        'SomeClass',
+                        'extend',
+                        1,
+                        0,
+                        0,
+                        true,
+                    ),
+                ]
+            ),
+            new ConfiguredCodeSample(
+                <<<'CODE_SAMPLE'
+$app->tag($stringVar, function ($parameter) {});
+CODE_SAMPLE
+                ,
+                <<<'CODE_SAMPLE'
+$app->tag($stringVar, function (string $parameter) {});
+CODE_SAMPLE
+                ,
+                [
+                    new AddParamTypeForFunctionLikeWithinCallLikeArgFromArgDeclaration(
+                        'SomeClass',
+                        'tag',
+                        1,
+                        0,
+                        0,
+                        false,
+                    ),
+                ]
+            ),
+        ]);
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [MethodCall::class, StaticCall::class];
+    }
+
+    /**
+     * @param CallLike $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        $this->hasChanged = false;
+        foreach ($this->addParamTypeForFunctionLikeParamDeclarations as $addParamTypeForFunctionLikeParamDeclaration) {
+            $type = match (true) {
+                $node instanceof MethodCall => $node->var,
+                $node instanceof StaticCall => $node->class,
+                default => null,
+            };
+
+            if ($type === null) {
+                continue;
+            }
+
+            if (! $this->isObjectType($type, $addParamTypeForFunctionLikeParamDeclaration->getObjectType())) {
+                continue;
+            }
+
+            if (($node->name ?? null) === null) {
+                continue;
+            }
+
+            if (! $node->name instanceof Identifier) {
+                continue;
+            }
```

Comment:
```suggestion
            if (! $node->name instanceof Identifier) {
                continue;
            }
```

I think this has the same result

Reply:
This fails PHPStan instead I've done:

```php
if (! ($node->name ?? null) instanceof Identifier) {
    continue;
}
```